### DISCUSSION TITLE
What does the Detector CPU refers to

### DISCUSSION THREAD

**Discussion by toperichvania** [blakeblackshear/frigate#14864] [Ask A Question] [answered]
I am wondering how to interpret the data from General Stats page, specifically high Detector CPU usage showing 97.3% utlization. But when I look at stats from nvidia-smi from container I get:

<img width="646" alt="Screenshot 2024-11-08 at 14 01 07" src="https://github.com/user-attachments/assets/5210b217-3618-4fdd-a687-d57377491f15">

and more or less the same from the OS:
<img width="643" alt="Screenshot 2024-11-08 at 14 00 50" src="https://github.com/user-attachments/assets/6adf1ac9-ee95-4152-9966-72b67e468e00">

My htop shows following:
<img width="866" alt="Screenshot 2024-11-08 at 14 02 18" src="https://github.com/user-attachments/assets/3469ef94-e4ba-4dcf-8cae-98dbd0530e4e">

but my system page on frigate shows this:
<img width="1503" alt="Screenshot 2024-11-08 at 13 57 11" src="https://github.com/user-attachments/assets/d1db529b-db4c-4a67-866f-7bff38702503">

**Comment by NickM-27** [comment]
it means there is a lot of CPU usage in the process responsible for running detection. Usually this is due to scaling the image and changing the pixel format

**Comment by toperichvania** [comment]
is there something I can do about this (hardware or config wise)?

**Reply by NickM-27** [reply]
would need to see the camera system stats page

**Comment by toperichvania** [comment]
<img width="1502" alt="Screenshot 2024-11-08 at 14 13 06" src="https://github.com/user-attachments/assets/a1e95316-a944-48fd-b2f6-a4d5c99ec7e6">
<img width="1494" alt="Screenshot 2024-11-08 at 14 13 30" src="https://github.com/user-attachments/assets/a15870ea-1875-4f26-a09c-6b57a116ce49">
<img width="1502" alt="Screenshot 2024-11-08 at 14 13 37" src="https://github.com/user-attachments/assets/59d48f28-0a3e-42c8-a746-49cff7ecc052">
<img width="1496" alt="Screenshot 2024-11-08 at 14 13 47" src="https://github.com/user-attachments/assets/3b3e1d7c-e79c-4dbf-86c6-61fe5d240f90">
<img width="1498" alt="Screenshot 2024-11-08 at 14 13 58" src="https://github.com/user-attachments/assets/76fc30e8-e0b9-43d6-a659-16c1d7c7eb61">
<img width="1496" alt="Screenshot 2024-11-08 at 14 14 06" src="https://github.com/user-attachments/assets/a57b7f93-ff38-49b1-8187-a5b2f720d226">
<img width="1499" alt="Screenshot 2024-11-08 at 14 14 15" src="https://github.com/user-attachments/assets/facbf622-8d37-4115-ad22-80a2e58af029">
<img width="1498" alt="Screenshot 2024-11-08 at 14 14 25" src="https://github.com/user-attachments/assets/a5259b77-4ac9-4288-9795-23ec671b0e45">
<img width="1503" alt="Screenshot 2024-11-08 at 14 14 33" src="https://github.com/user-attachments/assets/e292f8c7-d2f3-4c73-a23d-a97914d3f2da">
<img width="1505" alt="Screenshot 2024-11-08 at 14 14 43" src="https://github.com/user-attachments/assets/2c81320b-77c4-4031-b6e5-9dab40d8ce95">

**Reply by NickM-27** [reply, marked as answer]
some of these cameras have a lot of detections, I would suggest tuning motion detection in the UI. Also, md rescue has very high ffmpeg usage, perhaps reduce the detect resolution used there

**Comment by toperichvania** [comment]
perfect! Thanks for your help!!! I will play with those settings.

**Comment by toperichvania** [comment]
those two md_rescue cameras are actually 360 cameras with 2992x2992. I will use second stream for detection and see how that will affect it

**Reply by NickM-27** [reply]
you can always reduce the detect width / height while using the main stream, the resize will happen using gpu

**Comment by toperichvania** [comment]
oh, ok.. that is probably less on resources then having a second stream

**Comment by toperichvania** [comment]
but hypothetically, if I would need to add more cameras in the system that is already on high detection CPU, how would I go about it with hardware (if even possible). Would I need more CPU's or more power-full CPU's?
